src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1421,8 +1421,6 @@ void GenerateResetMessage()
         /// </remarks>
         internal void Reset()
         {
-            Debug.Assert(State == ConnectorState.Ready);
-
```

Comment:
Looks unrelated to the PR.

Reply:
This was triggered on Github Actions at some point (also marked the test as flaky). In any case the assert is likely false - some race conditions may cause Reset to be called twice, in which case it's already closed/broken (see XML comment above the function). We also check for everything in the switch right below.